nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/proto/ProtoUtils.java

```diff
@@ -46,7 +48,21 @@ public static DynamicMessage createMessage(Descriptors.Descriptor descriptor, Ma
            switch (field.getType()) {
            case MESSAGE:
                if (field.isRepeated()) {
-                   Collection collection = value.getClass().isArray() ? Arrays.asList((Object[]) value) : (Collection) value;
+                   Collection collection = null;
+                   if (value.getClass().isArray()) {
+                       collection = Arrays.asList((Object[]) value);
+                   } else if (value instanceof HashMap) {
+                       collection = new ArrayList();
+                       for (Object entryObj : ((HashMap<?, ?>) value).entrySet()) {
+                           Map.Entry<?, ?> entry = (Map.Entry<?, ?>) entryObj;
+                           Map<String, Object> map = new HashMap<>();
+                           map.put("key", entry.getKey());
+                           map.put("value", entry.getValue());
+                           collection.add(map);
+                       }
+                   } else {
+                       collection = (Collection) value;
+                   }
                    collection.forEach(act -> builder.addRepeatedField(field, createMessage(field.getMessageType(), (Map<String, Object>) act, tableSchema)));
```

Comment:
As I've changed the type of the former `collection` field, the unchecked cast must be altered slightly.

However, we can drop the cast inside the `foreach` this way.

```suggestion
                   } else {
                       valueMaps = (Collection<Map<String, Object>>) value;
                   }

                   valueMaps.forEach(act -> builder.addRepeatedField(field, createMessage(field.getMessageType(), act, tableSchema)));
```